sdk/common/src/main/java/io/opentelemetry/sdk/resources/ResourceProvider.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.resources;
+
+/** A Registry which provides the {@link Resource} to the SDK. */
+public final class ResourceProvider {
```

Comment:
We already have ResourceProvider and will likely need to pick a different name.

Separately, I'm a little confused on the purpose of this thing. Seems like we could get away extending ResourceBuilder to include entity concepts.

Reply:
It's called out in the OTEP: https://github.com/open-telemetry/opentelemetry-specification/blob/main/oteps/entities/0264-resource-and-entities.md#resource-provider

Initially it doesn't do much, eventually it'll be critical for client-side handling (e.g. Android SDK)